samtranslator/model/iam.py

```diff
@@ -129,3 +129,45 @@ def lambda_invoke_function_role_policy(cls, function_arn, logical_id):
             },
         }
         return document
+
+    @classmethod
+    def execute_canary_policy(cls, logical_id, result_bucket, canary_name):
+        document = {
+            "PolicyName": logical_id + "CanaryPolicy",
+            "PolicyDocument": {
+                "Statement": [
+                    {
+                        "Effect": "Allow",
+                        "Action": ["s3:PutObject", "s3:GetBucketLocation"],
+                        "Resource": [{"Fn::Join": ["", ["arn:aws:s3:::", result_bucket, "/*"]]}],
+                    },
+                    {
+                        "Effect": "Allow",
+                        "Action": ["logs:CreateLogStream", "logs:PutLogEvents", "logs:CreateLogGroup"],
+                        "Resource": [
+                            {
+                                "Fn::Join": [
+                                    "",
+                                    [
+                                        {
+                                            "Fn::Sub": "arn:${AWS::Partition}:logs:${AWS::Region}:${"
+                                            "AWS::AccountId}:log-group:/${AWS::Partition}/lambda/cwsyn-"
+                                        },
+                                        canary_name,
+                                        "-*",
+                                    ],
+                                ]
+                            }
+                        ],
+                    },
+                    {"Effect": "Allow", "Action": ["s3:ListAllMyBuckets"], "Resource": ["*"]},
+                    {
+                        "Effect": "Allow",
+                        "Action": "cloudwatch:PutMetricData",
+                        "Resource": "*",
```

Comment:
Is there a way to be more specific about the resource? (or the condition below is enough)

Reply:
the condition should be enough, its the same one used by cloudwatch dashboard when you generate a role policy through there

Reply:
I don't think that is Sam meant about Resources. To answer Sam's question we allow Canary Lambda to putmetricdata so that the integration to publishing metrics is not very brittle. Another approach could be that metric data is created and only permission to those will be granted. But every time we require to publish new metric we have to update it. Plus I couldn't find any way to create metric data using CFN, so it might have to be done by hand. This is similar to `LambdaBasicExecution` permission for logs.